align/schema/constraint.py

```diff
@@ -516,9 +517,13 @@ class GroupBlocks(HardConstraint):
                             }
                         }
         }
+
+    Note: To keep backward compatibility with previous version, in case template name is not provided, name is used as template name and  instance name is generated based on the combination of "name" and the list of instance names (e.g. X_GROUP1_MN0_MN1_MN3 for the example below).
```

Comment:
@kkunal1408 , to make it consistent, why not use `name` as instance name always and generate `template` name if not provided?

Reply:
`name` could be used in other placement constraints and this would avoid any translations in the constraints.

Reply:
@kkunal1408 , let's rename `name` as `instance_name` to prevent confusion. Whatever specifies the grouping, `template_name` should be unique, and `instance_name` should be unique within circuit. Both of these should be easily verifiable thru the flow. 

To accommodate the subcircuit prefix requirement, let's require `instance_name` to start with 'x' using a pydantic validator.   

If not specified by the user, I do not have a strong opinion on how `template_name` is generated as long as it is unique and robust. If specified by the user, the flow should assert for uniquness.

Although 40+ files sounds scary, modifying and reviewing these lines will not be hard. We can split the work. I can help with `tests/pdk/finfet` to start with.

Reply:
Sure. I will update the relevant files with new naming convention of group_block as suggested.

Reply:
@soneryaldiz  fixed